sherpa/astro/ui/utils.py

```diff
@@ -2525,7 +2577,9 @@ def load_filter(self, id, filename=None, bkg_id=None, ignore=False,
     # DOC-TODO: prob. needs a review as the existing ahelp documentation
     # talks about 2 cols, but experimentation suggests 1 col.
     #
-    def load_grouping(self, id, filename=None, bkg_id=None, *args, **kwargs):
+    def load_grouping(self, id, filename=None,
```

Comment:
Inconsisteny with the doctring, which says filename is a str.

Reply:
Same happens below in different locations where the other parameters of a function are typed in this PR. Again, I want to discuss here and not mark all instances at this stage.

Reply:
I point your attention to the "Notes" section of this (and many other routines in this module):

https://github.com/sherpa/sherpa/blob/307ddbbd4b7ff98e3c9720915f05aa530f4a1262/sherpa/astro/ui/utils.py#L2528-L2576

I have been worrying about how to type these, or whether we even can, for a long time now. However, that was before I knew about [typing.overload](https://docs.python.org/3/library/typing.html#overload). I now think we could use that approach, but

- I have not tied it
- it would be massively invasive, so I don't think now is the time
- it looks like the semantics of `overload` is not entirely resolved - see the [discourse discussion](https://discuss.python.org/t/draft-typing-spec-chapter-for-overloads/60788) - which is always fun to see (although I'm not sure we are likely to hit on these areas)

Reply:
Oh, I hate these "designed for easy interactive use". It's easy - unless you want to see what you need to put in. But we're not going to redesign the UI because I don't like it.
OK, I'll live with it.